psa-crypto-sys/build.rs

```diff
@@ -195,84 +98,49 @@ fn generate_mbed_bindings(mbed_config: &MbedConfig, mbed_version: &str) -> Resul
-fn parse_config_file() -> Result<Configuration> {
-    let config_str = ::std::fs::read_to_string(Path::new(BUILD_CONFIG_FILE_PATH))?;
-    Ok(toml::from_str(&config_str).or_else(|e| {
-        println!("Error parsing build configuration file ({}).", e);
-        Err(Error::new(
-            ErrorKind::InvalidInput,
-            "Could not parse build configuration file.",
-        ))
-    })?)
+    Ok(())
 }
 
-fn main() -> Result<()> {
-    // Parsing build-conf.toml
-    let config = parse_config_file()?;
-
-    // Parsing Cargo.toml
-    let toml_path = std::path::Path::new("./Cargo.toml");
-    if !toml_path.exists() {
-        return Err(Error::new(
-            ErrorKind::InvalidInput,
-            "Could not find Cargo.toml.",
-        ));
-    }
-    let manifest = Manifest::from_path(&toml_path).or_else(|e| {
-        println!("Error parsing Cargo.toml ({}).", e);
-        Err(Error::new(
-            ErrorKind::InvalidInput,
-            "Could not parse Cargo.toml.",
-        ))
-    })?;
-
-    let package = manifest.package.ok_or_else(|| {
-        Error::new(
-            ErrorKind::InvalidInput,
-            "Cargo.toml does not contain package information.",
-        )
-    })?;
-    let metadata = package.metadata.ok_or_else(|| {
-        Error::new(
-            ErrorKind::InvalidInput,
-            "Cargo.toml does not contain package metadata.",
-        )
-    })?;
-    let parsec_config = get_value_from_table(&metadata, CONFIG_TABLE_NAME)?;
+fn link_to_lib(lib_path: String) -> Result<()> {
+    let link_type = if cfg!(feature = "dynamic") {
+        "dylib"
+    } else {
+        "static"
+    };
 
-    if true {
-        let mbed_config = config.mbed_config.ok_or_else(|| {
-            Error::new(
-                ErrorKind::InvalidInput,
-                "Could not find mbed_config table in the config file.",
-            )
-        })?;
+    // Request rustc to link the Mbed Crypto library
+    println!("cargo:rustc-link-search=native={}", lib_path,);
+    println!("cargo:rustc-link-lib={}=mbedcrypto", link_type);
 
-        let mbed_version = get_configuration_string(&parsec_config, MBED_CRYPTO_VERSION_KEY)?;
+    // Also link shim library
+    println!(
+        "cargo:rustc-link-search=native={}",
+        env::var("OUT_DIR").unwrap()
+    );
+    println!("cargo:rustc-link-lib={}=shim", link_type);
 
-        setup_mbed_crypto(&mbed_config, &mbed_version)?;
-        generate_mbed_bindings(&mbed_config, &mbed_version)?;
+    Ok(())
+}
 
-        // Request rustc to link the Mbed Crypto static library
-        println!(
-            "cargo:rustc-link-search=native={}/mbed-crypto-{}/library/",
-            mbed_config
-                .mbed_path
-                .unwrap_or_else(|| env::var("OUT_DIR").unwrap()),
-            mbed_version,
-        );
-        println!("cargo:rustc-link-lib=static=mbedcrypto");
+fn main() -> Result<()> {
```

Comment:
The logic is much simpler and nicer!
I have been reading [the article](https://kornel.ski/rust-sys-crate) again and I think we should follow this decision path:

1. Is the library available? (We should ideally use `pkg-config` for that but currently use the env variable as you did). "Yes" go to 2, "No" go to 3.
2. Link with the library _dynamically_ by default or statically if the `static` feature is set
3. Compile the library from the submodule and link with it **always statically**.

I think that libraries that we compile ourselves should always be linked statically but we leave the option for libraries that are available on the system.